server/src/main/java/org/elasticsearch/search/aggregations/metrics/HDRPercentilesAggregator.java

```diff
@@ -55,6 +55,6 @@ public double metric(String name, long bucketOrd) {
 
     @Override
     public InternalAggregation buildEmptyAggregation() {
-        return new InternalHDRPercentiles(name, keys, null, keyed, format, metadata());
+        return new InternalHDRPercentiles(name, keys, new DoubleHistogram(numberOfSignificantValueDigits), keyed, format, metadata());
```

Comment:
Note that in `TDigestPercentilesAggregator#buildEmptyAggregation` we do something similar

```
return new InternalTDigestPercentiles(name, keys, new TDigestState(compression), keyed, formatter, metadata());
```

Reply:
Creating `DoubleHistogram` with a high `numberOfSignificantValueDigits`  consumes a lot of memory. Can we avoid doing this here? Also serialising it would be non trivial.

Can we maybe create the right `DoubleHistogram ` instance in `AbstractInternalHDRPercentiles#reduce(...)` method? Right now it always uses a `EMPTY_HISTOGRAM` in case of an empty bucket. Maybe we create another instances if `numberOfSignificantValueDigits` != 3?